test/linalg/rowvector.jl

```diff
@@ -249,3 +249,7 @@ end
         @test A'*x' == A'*y == B*x' == B*y == C'
     end
 end
+
+@testset "ambiguity between * methods with RowVectors and ConjRowVectors (#20971)" begin
+    @test_nowarn RowVector(ConjArray(rand(4))) * rand(4)
```

Comment:
I'd do a real example and check its result, rather than nowarn on random data - ambiguities should be errors now